assets/tests/hooks/useDetour.test.ts

```diff
@@ -0,0 +1,132 @@
+import { beforeEach, describe, expect, jest, test } from "@jest/globals"
+import { fetchDetourDirections } from "../../src/api"
+import { renderHook, waitFor } from "@testing-library/react"
+import { useDetour } from "../../src/hooks/useDetour"
+import { act } from "react-dom/test-utils"
+
+jest.mock("../../src/api")
+
+beforeEach(() => {
+  jest.mocked(fetchDetourDirections).mockResolvedValue(null)
+})
+
+describe("useDetour", () => {
+  test("when `addConnectionPoint` is first called, `startPoint` is set", () => {
+    const start = { lat: 0, lon: 0 }
+
+    const { result } = renderHook(useDetour)
+
+    act(() => result.current.addConnectionPoint(start))
+
+    expect(result.current.startPoint).toBe(start)
+  })
+
+  test("when `addConnectionPoint` is called a second time, `endPoint` is set", () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    const { result } = renderHook(useDetour)
+
+    act(() => result.current.addConnectionPoint(start))
+    act(() => result.current.addConnectionPoint(end))
+
+    expect(result.current.startPoint).toBe(start)
+    expect(result.current.endPoint).toBe(end)
+  })
+
+  test("when `startPoint` is null, `addWaypoint` does nothing", () => {
+    const { result } = renderHook(useDetour)
+
+    expect(result.current.startPoint).toBeNull()
+
+    act(() => result.current.addWaypoint({ lat: 0, lon: 0 }))
+
+    expect(result.current.waypoints).toEqual([])
+  })
+
+  test("when `endPoint` is set, `addWaypoint` does nothing", () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    const { result } = renderHook(useDetour)
+
+    expect(result.current.startPoint).toBeNull()
+
+    act(() => result.current.addConnectionPoint(start))
+    act(() => result.current.addConnectionPoint(end))
+
+    act(() => result.current.addWaypoint({ lat: 0, lon: 0 }))
+
+    expect(result.current.waypoints).toEqual([])
+  })
+
+  test("when `addWaypoint` is called, `detourShape` is updated", async () => {
+    const start = { lat: 0, lon: 0 }
+    const end = { lat: 1, lon: 1 }
+
+    jest
+      .mocked(fetchDetourDirections)
+      .mockImplementation((coordinates) =>
+        Promise.resolve({ coordinates: coordinates })
+      )
```

Comment:
Nit: This test would pass if the [`useEffect` in `useDetourDirections`](https://github.com/mbta/skate/pull/2364/files#diff-ea299f4c918f1648d9f7ab5eb825737dd5e598ed5f3d3a9db036ae03d447d224R8-R24) just ignored the API and returned the input. 

I'd recommend changing it to something like 👇, along with the associated 

```suggestion
    jest
      .mocked(fetchDetourDirections)
      .mockImplementation((coordinates) =>
        Promise.resolve({ coordinates: [
          { lat: 0, lon: 0 },
          { lat: 1, lon: 0},
          { lat: 1, lon: 1},
        ] })
      )
```

Reply:
baef5e3196b0a44cd57d1a515c7c2c765f5214bb